src/js/components/DataTable/DataTable.js

```diff
@@ -131,6 +131,16 @@ const DataTable = ({
   const [headerHeight, setHeaderHeight] = useState();
   const [footerHeight, setFooterHeight] = useState();
 
+  // offset compensation when body overflows
+  const [scrollOffset, setScrollOffset] = useState(0);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const bodyRef = useCallback(ref => {
```

Comment:
One suggestion, can we do something like this instead?
```
  const bodyRef = useRef();
  useLayoutEffect(() => {
    const nextScrollOffset =
      bodyRef.current.parentElement?.clientWidth - bodyRef.current.clientWidth;
    if (nextScrollOffset !== scrollOffset) setScrollOffset(nextScrollOffset);
  });

Reply:
Sure thing! I'll push a commit addressing that! Also, thanks for pointing that out. Never had a use case for useLayoutEffect so I've been reading about it a little bit 😄